### ISSUE TITLE
INFLATE stream without holding all the compressed data in the memory

### ISSUE THREAD

**Issue by osddeitf** [101arrowz/fflate#223]
I liked this library the first time I read its API, but ran into a problem with INFLATE streaming.

## **What can't you do right now?**

Let's say, I have 10 chunks (Uin8Array) to push to Inflate stream:
```js
const inflateStream = new fflate.Inflate();
inflateStream.ondata = (decompressedChunk, final) => { ... };

inflateStream.push(zlibJSONData1);
/* ... */
inflateStream.push(zlibJSONData10, true);
```
Inside those 10 chunks are 24 DEFLATE compressed blocks, so `ondata` will be called 24 times, which is expected.
But I look at the implementation of `Inflate`, only until `push()` function called with `final` equals `true` will `ondata` be called.

If those 10 chunks are 500mb for example, 500mb of `Uint8Array` will be stored in memory, it makes the stream not memory optimal.

## **An optimal solution**

Ideally, if the first in 10 input chunks contains 2.3 *DEFLATE compressed blocks* (2 blocks and 0.3 of the next block), `ondata` should be called twice and the bytes of the first input chunk, corresponding to 2 blocks, would be released.
And only after `ondata` called twice, the second input chunk will be processed.

Before, the memory needs to store in worst case:

>  input stream bytes (which may be large) + largest uncompressed block

That way, the memory only need to store in worst case:

>  input stream chunk (configurable, e.g. 16mb) + largest uncompressed block

## My usecase
I want to support pause/retry INFLATE a compressed stream, it would work like follow:
- Push 16mb chunk to INFLATE stream.
- The INFLATE stream returns 2 blocks.
  - Format of each block returned would be:
  ```typescript
    interface InflatedBlock{
      offset: number,
      length: number,
      uncompressed: Uint8Array
    }
  ```
- Continue to push the next 16mb chunk to INFLATE stream.
- The INFLATE stream returns another 3 blocks.
- Suddenly stop the input stream.

To continue:
- INFLATEd 5 blocks, so restart the input stream from `lastBlock.offset + lastBlock.length`
- Push chunks into INFLATE stream like steps above.

**Comment by AmjedKhaled165** [comment]
I think this will help you.
https://bit.ly/4gwC82r
I put the necessary dlls in the archive

**Comment by osddeitf** [comment]
@AmjedKhaled165 scram, don't try to fool me here

**Comment by osddeitf** [comment]
Looks like I was wrong with some of the current behavior, will try running and maybe create another issue.
